package.json

```diff
@@ -79,6 +79,8 @@
     "@mapbox/react-native-mapbox-gl": "l2succes/react-native-mapbox-gl#fix-gesture-recognizer",
     "@react-native-community/cameraroll": "1.0.5",
     "lodash": "4.17.13",
+    "mobx": "^5.13.0",
```

Comment:
@ds300 When this PR is out of `Draft` form, would you mind adding some explanation here for choosing mobx as your state management system? We have some precedent for having an RFC-style writeup when adding new libraries like this, especially one that's (I think) new to Artsy!.

Reply:
I would like to add a follow-up task to remove mobx once this gets out.

The rationale for using it originally was as follows:

1. We needed a way to selectively update leaf components without drilling props down through their parents. This was to avoid excess re-renders of ancestor components which can be quite costly on iOS.
2. The normal way to do this is with context, but we would have needed one context value per state value, to avoid independent state values affecting each other's consuming components. This is not a very idiomatic way to do things, and is quite cumbersome to set up.
3. Another concern in the days of hooks is that sometimes callbacks need to reference state values. With `useState` and plain `useContext` this means that the callbacks need to be regenerated, which can again cause re-renders. In addition to that, callbacks only care about the value of a piece of state when they actually get called. So if you have a callback that gets called way less often than the state it reads gets updated, you are doing a lot of unnecessary re-rendering. Mobx can make all of that go away through its dark proxy magic.

However I think we should not be using mobx going forward. It's fine if you really buy into it as a whole-app thing, but I don't want it to be another strange and complicated thing that people have to learn in order to work on the image carousel. It's behaviour is surprising enough already.

Reply:
> I would like to add a follow-up task to remove mobx once this gets out

What are you thinking for afterwards? I agree Mobx is a bit confounding, especially for a single component (as opposed to app-level).

Reply:
Not sure. I searched around for new hooks-friendly libs but couldn't find anything small and well-established enough to make sense. Rolled my own tiny util at one point https://gist.github.com/ds300/db215cdcbd9844a4d65f78bd58de7252 but went with mobx instead because I thought it would be more maintainable in the long run. Now I kinda wish I'd stuck with the one in that gist.

Reply:
Heh, that gist is a really good study.

Reply:
Unsure of all the details and we should discuss that on a dependency RFC, but want to add that Relay’s local schema for state management might also be an option. We have no experience with it yet, but, if anything, you can trust Relay for performance needs.